src/main/java/com/aws/greengrass/config/ConfigurationWriter.java

```diff
@@ -198,13 +209,15 @@ public void writeAll() {
     /**
      * Create a new Writer for writing to a tlog file.
      *
-     * @param outputPath path to tlog file
+     * @param outputPath
+     *            path to tlog file
      * @return a new writer
-     * @throws IOException if I/O error creating output file or writer
+     * @throws IOException
+     *             if I/O error creating output file or writer
      */
     private static Writer newTlogWriter(Path outputPath) throws IOException {
-        return Files.newBufferedWriter(outputPath, StandardOpenOption.APPEND,
-                StandardOpenOption.SYNC, StandardOpenOption.CREATE);
+        return Files.newBufferedWriter(outputPath, StandardOpenOption.APPEND, StandardOpenOption.SYNC,
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Path traversal vulnerability detected. User-controlled input in file paths
can allow attackers to access files outside intended directories using
`../` sequences. Secure your code by using framework functions like
`getCanonicalPath()`, `normalize()`, or validating paths with
`.startsWith()` checks. Learn more:
https://cwe.mitre.org/data/definitions/22.html